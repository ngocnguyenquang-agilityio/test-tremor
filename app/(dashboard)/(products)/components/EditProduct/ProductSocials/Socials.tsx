"use client";

import { Controller, useFormContext } from "react-hook-form";

// Components
import { Title, Flex, Card } from "@tremor/react";
import { InputField } from "@/components";

//Styles
import "@/styles/products.css";

// Constants
import { MESSAGES_ERROR, REGEX } from "@/constants";

// Hooks
import useFocusFieldError from "@/hooks/useFocusFieldError";

const Socials = () => {
  const { control, formState } = useFormContext();

  useFocusFieldError(formState);

  return (
    <Card className="dark:bg-dark-tremor-primary ring-0">
      <Title className="font-primary font-bold text-primary dark:text-dark-primary text-xl leading-snug capitalize">
        Socials
      </Title>
      <Flex className="flex-col items-start justify-start mt-4 gap-6">
        <Controller
          control={control}
          rules={{
            pattern: {
              value: REGEX.URL,
              message: MESSAGES_ERROR.INVALID_URL,
            },
          }}
          render={({ field, formState: { errors } }) => {
            const shopifyUrlErrorMessage = errors.shopifyUrl?.message || "";

            return (
              <div className="w-full mb-4">
                <InputField
                  id="edit-shopify"
                  label="Shopify Handle"
                  type="text"
                  errorMessage={shopifyUrlErrorMessage}
                  {...field}
                />
              </div>
            );
          }}
          name="shopifyUrl"
        />
        <Controller
          control={control}
          rules={{
            pattern: {
              value: REGEX.URL,
              message: MESSAGES_ERROR.INVALID_URL,
            },
          }}
          render={({ field, formState: { errors } }) => {
            const facebookUrlErrorMessage = errors.facebookUrl?.message || "";

            return (
              <div className="w-full mb-4">
                <InputField
                  id="edit-fb"
                  label="Facebook Account"
                  type="text"
                  errorMessage={facebookUrlErrorMessage}
                  {...field}
                />
              </div>
            );
          }}
          name="facebookUrl"
        />
        <Controller
          control={control}
          rules={{
            pattern: {
              value: REGEX.URL,
              message: MESSAGES_ERROR.INVALID_URL,
            },
          }}
          render={({ field, formState: { errors } }) => {
            const instagramUrlErrorMessage = errors.instagramUrl?.message || "";

            return (
              <div className="w-full mb-4">
                <InputField
                  id="edit-ig"
                  label="Instagram Account"
                  type="text"
                  errorMessage={instagramUrlErrorMessage}
                  {...field}
                />
              </div>
            );
          }}
          name="instagramUrl"
        />
      </Flex>
    </Card>
  );
};

export default Socials;
