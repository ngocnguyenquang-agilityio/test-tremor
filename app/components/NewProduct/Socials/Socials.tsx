// Components
import { Title, Flex } from "@tremor/react";

const Socials = () => {
  const inputClasses = `w-full text-sm text-tremor-content-title py-1 mb-6 rounded-none ring-0 border-0 border-b focus:border-b-2 border-gray-300 focus:border-tremor-brand-subtle focus:outline-none hover:bg-transparent dark:bg-transparent dark:hover:bg-transparent dark:border-white`;
  return (
    <>
      <Title className="font-primary font-bold text-primary dark:text-dark-primary text-xl leading-snug capitalize">
        Socials
      </Title>
      <Flex className="flex-col items-start justify-start mt-8 text-secondary dark:text-dark-romance text-sm font-normal">
        <input
          className={inputClasses}
          id="shoppifyHandle"
          placeholder="Shoppify Handle"
        />
        <input
          className={inputClasses}
          id="facebookAccount"
          placeholder="Facebook Account"
        />
        <input
          className={inputClasses}
          id="instagramAccount"
          placeholder="Instagram Account"
        />
      </Flex>
    </>
  );
};

export default Socials;
