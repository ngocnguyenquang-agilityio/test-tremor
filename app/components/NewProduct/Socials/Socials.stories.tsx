import type { Meta, StoryObj } from "@storybook/react";

// Components
import Socials from "./Socials";

const meta = {
  title: "Components/Socials",
  component: Socials,
  tags: ["autodocs"],
} as Meta<typeof Socials>;

export default meta;

type Story = StoryObj<typeof meta>;

const propsDefault = {
  handleSocials: () => {},
};

export const Primary: Story = {
  render: () => <Socials {...propsDefault} />,
};
