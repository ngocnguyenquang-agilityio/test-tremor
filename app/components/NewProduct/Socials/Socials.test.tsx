import { render } from "@testing-library/react";

// Components
import Socials from "./Socials";

describe("Testing Order detail contact section", () => {
  it("Should match snapshot", () => {
    const component = render(<Socials />);
    expect(component).toMatchSnapshot();
  });
});
