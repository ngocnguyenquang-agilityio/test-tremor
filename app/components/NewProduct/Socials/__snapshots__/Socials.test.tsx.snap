// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Order detail contact section Should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="text-tremor-title font-primary font-bold text-primary dark:text-dark-primary text-xl leading-snug capitalize"
      >
        Socials
      </p>
      <div
        class="tremor-Flex-root flex w-full flex-col items-start justify-start mt-8"
      >
        <input
          class="w-full text-sm text-secondary dark:text-dark-romance text-tremor-content-title py-1 mb-6 rounded-none ring-0 border-0 border-b focus:border-b-2 border-gray-300 focus:border-tremor-brand-subtle focus:outline-none hover:bg-transparent dark:bg-transparent dark:hover:bg-transparent dark:border-white dark:text-white"
          id="shoppifyHandle"
          placeholder="Shoppify Handle"
        />
        <input
          class="w-full text-sm text-secondary dark:text-dark-romance text-tremor-content-title py-1 mb-6 rounded-none ring-0 border-0 border-b focus:border-b-2 border-gray-300 focus:border-tremor-brand-subtle focus:outline-none hover:bg-transparent dark:bg-transparent dark:hover:bg-transparent dark:border-white dark:text-white"
          id="facebookAccount"
          placeholder="Facebook Account"
        />
        <input
          class="w-full text-sm text-secondary dark:text-dark-romance text-tremor-content-title py-1 mb-6 rounded-none ring-0 border-0 border-b focus:border-b-2 border-gray-300 focus:border-tremor-brand-subtle focus:outline-none hover:bg-transparent dark:bg-transparent dark:hover:bg-transparent dark:border-white dark:text-white"
          id="instagramAccount"
          placeholder="Instagram Account"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <p
      class="text-tremor-title font-primary font-bold text-primary dark:text-dark-primary text-xl leading-snug capitalize"
    >
      Socials
    </p>
    <div
      class="tremor-Flex-root flex w-full flex-col items-start justify-start mt-8"
    >
      <input
        class="w-full text-sm text-secondary dark:text-dark-romance text-tremor-content-title py-1 mb-6 rounded-none ring-0 border-0 border-b focus:border-b-2 border-gray-300 focus:border-tremor-brand-subtle focus:outline-none hover:bg-transparent dark:bg-transparent dark:hover:bg-transparent dark:border-white dark:text-white"
        id="shoppifyHandle"
        placeholder="Shoppify Handle"
      />
      <input
        class="w-full text-sm text-secondary dark:text-dark-romance text-tremor-content-title py-1 mb-6 rounded-none ring-0 border-0 border-b focus:border-b-2 border-gray-300 focus:border-tremor-brand-subtle focus:outline-none hover:bg-transparent dark:bg-transparent dark:hover:bg-transparent dark:border-white dark:text-white"
        id="facebookAccount"
        placeholder="Facebook Account"
      />
      <input
        class="w-full text-sm text-secondary dark:text-dark-romance text-tremor-content-title py-1 mb-6 rounded-none ring-0 border-0 border-b focus:border-b-2 border-gray-300 focus:border-tremor-brand-subtle focus:outline-none hover:bg-transparent dark:bg-transparent dark:hover:bg-transparent dark:border-white dark:text-white"
        id="instagramAccount"
        placeholder="Instagram Account"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
